mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/spi/PojoMassIndexer.java

```diff
@@ -130,4 +131,20 @@ public interface PojoMassIndexer {
 	 * @return {@code this} for method chaining
 	 */
 	PojoMassIndexer failureHandler(MassIndexingFailureHandler failureHandler);
+
+	/**
+	 * Sets the maximum number of failures that will be reported and sent to {@link MassIndexingFailureHandler} per indexing operation.
+	 * Any failures exceeding this number will be ignored. A count of such ignored failures together with the operation they belong to
+	 * will be logged upon the completion of indexing process.
+	 * <p>
+	 * By default, this number is set to {@code 1000}.
+	 *
+	 * @param maximumFailuresHandledPerOperation The number representing the maximum failures to be handled per operation.
+	 * @return {@code this} for method chaining
+	 */
+	@Incubating
+	default PojoMassIndexer maximumFailuresHandledPerOperation(long maximumFailuresHandledPerOperation) {
+		throw new UnsupportedOperationException(
+				this.getClass() + " does not support setting the `maximumFailuresHandledPerOperation`. Implement it first." );
+	}
```

Comment:
No need for a default method here, this SPI is expected to be *called* by mappers, not implemented. 

I see how the name `PojoDefaultMassIndexer` makes that unclear, it should probably be called `PojoMassIndexerImpl`. But, well, that's not our concern right.